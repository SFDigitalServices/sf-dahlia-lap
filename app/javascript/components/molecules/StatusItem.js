import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import StatusPill from '../atoms/StatusPill'

import StatusItemShape from '../../utils/shapes/StatusItemShape'
import { getSubStatusLabel } from '../../utils/statusUtils'

const StatusDate = ({ timestamp }) => (
  <div className='status-item-date'>
    {moment.unix(timestamp).format('MMM D, YYYY')}
  </div>
)

StatusDate.propTypes = { timestamp: PropTypes.number }

const StatusItem = ({ statusItem }) => {
  const { status, substatus, comment, timestamp } = statusItem

  const substatusLabel = getSubStatusLabel(status, substatus)

  return (
    <div className='status-item'>
      <div className='status-item-header'>
        <StatusPill status={status} />
        <StatusDate timestamp={timestamp} />
      </div>
      {substatusLabel && (
        <div className='item-substatus-text'>
          {substatusLabel}
        </div>
      )}
      <div className='status-item-text'>
        {comment}
      </div>
    </div>
  )
}

StatusItem.propTypes = {
  statusItem: PropTypes.shape(StatusItemShape).isRequired
}

export default StatusItem
