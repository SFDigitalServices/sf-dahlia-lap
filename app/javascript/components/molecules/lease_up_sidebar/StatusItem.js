import React from 'react'

import moment from 'moment'
import PropTypes from 'prop-types'

import StatusItemShape from '../../../utils/shapes/StatusItemShape'
import { getSubStatusLabel } from '../../../utils/statusUtils'
import StatusPill from '../../atoms/StatusPill'

const StatusDate = ({ timestamp }) => (
  <div className='status-item-date'>{moment.unix(timestamp).format('MMM D, YYYY')}</div>
)
const StatusCreatedBy = ({ createdBy }) => <div className='status-item-created-by'>{createdBy}</div>

StatusDate.propTypes = { timestamp: PropTypes.number }

const StatusItem = ({ statusItem }) => {
  const { status, substatus, comment, timestamp, created_by } = statusItem

  const substatusLabel = getSubStatusLabel(status, substatus)

  return (
    <div className='status-item'>
      <div className='status-item-header'>
        <StatusPill status={status} />
      </div>
      {substatusLabel && <div className='item-substatus-text'>{substatusLabel}</div>}
      <div className='status-item-text'>{comment}</div>
      <div className='status-item-footer'>
        <StatusCreatedBy createdBy={created_by} />
        <StatusDate timestamp={timestamp} />
      </div>
    </div>
  )
}

StatusItem.propTypes = {
  statusItem: PropTypes.shape(StatusItemShape).isRequired
}

export default StatusItem
