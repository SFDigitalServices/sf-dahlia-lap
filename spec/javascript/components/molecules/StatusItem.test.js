import React from 'react'
import { shallow } from 'enzyme'
import { mockStatusItem } from '../../mocks/statusItemMock'
import StatusItem from 'components/molecules/StatusItem'

const getWrapper = (item) => shallow(<StatusItem statusItem={item} />)

describe('StatusItem', () => {
  test('should render an approved status with substatus and comments correctly', () => {
    const wrapper = getWrapper(mockStatusItem())
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no substatus provided', () => {
    const wrapper = getWrapper(mockStatusItem({ substatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no comment provided', () => {
    const wrapper = getWrapper(mockStatusItem({ comment: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly when no comment or substatus provided', () => {
    const wrapper = getWrapper(mockStatusItem({ subStatus: null, comment: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with withdrawn status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Withdrawn', subStatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Processing status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Processing', subStatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Appealed status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Appealed', subStatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Disqualified status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Disqualified', subStatus: null }))
    expect(wrapper).toMatchSnapshot()
  })

  test('should render properly with Lease Signed status', () => {
    const wrapper = getWrapper(mockStatusItem({ status: 'Lease Signed', subStatus: null }))
    expect(wrapper).toMatchSnapshot()
  })
})
