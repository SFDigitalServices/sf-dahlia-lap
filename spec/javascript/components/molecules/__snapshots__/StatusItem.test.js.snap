// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusItem should render an approved status with substatus and comments correctly 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Approved"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="item-substatus-text"
  >
    Approval letter sent
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly when no comment or substatus provided 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Approved"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  />
</div>
`;

exports[`StatusItem should render properly when no comment provided 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Approved"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="item-substatus-text"
  >
    Approval letter sent
  </div>
  <div
    className="status-item-text"
  />
</div>
`;

exports[`StatusItem should render properly when no substatus provided 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Approved"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly with Appealed status 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Appealed"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly with Disqualified status 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Disqualified"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly with Lease Signed status 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Lease Signed"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly with Processing status 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Processing"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;

exports[`StatusItem should render properly with withdrawn status 1`] = `
<div
  className="status-item"
>
  <div
    className="status-item-header"
  >
    <StatusPill
      status="Withdrawn"
    />
    <StatusDate
      timestamp={1598385600}
    />
  </div>
  <div
    className="status-item-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque bibendum condimentum lorem consectetur eleifend.
  </div>
</div>
`;
